pipeline_steps/run_tests.groovy

```diff
@@ -0,0 +1,38 @@
+def run_irr_tests() {
+  common.conditionalStage(
+    stage_name: "Run IRR Tests",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          ansiColor('xterm') {
+            dir(env.WORKSPACE) {
+              print("Triggered by PR: ${env.ghprbPullLink}")
+              checkout([$class: 'GitSCM',
+                branches: [[name: "origin/pr/${env.ghprbPullId}/merge"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'CleanCheckout']],
+                submoduleCfg: [],
+                userRemoteConfigs: [
+                  [
+                    url: "https://github.com/${env.ghprbGhRepository}.git",
+                    refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
+                  ]
+                ]
+              ]) // checkout
+              sh """#!/bin/bash
+              pushd \$(basename ${env.ghprbGhRepository})
+                bash ./run_tests.sh
+              popd
+              """
+            } // dir
+          } // ansiColor
+        } catch (e) {
+          print(e)
+          throw e
+        } finally {
+          common.archive_artifacts()
```

Comment:
No harm in running it, more of a heads up that, `common.archive_artifacts` is not a general artifact collection function, it is specifically aimed at collecting logs and config from rpc deploys, so additional archiving may be required for tests that aren't a straightforward rpc deploy.